safenode/src/domain/storage/disk_backed_record_store.rs

```diff
@@ -228,13 +247,10 @@ impl RecordStore for DiskBackedRecordStore {
     }
 
     fn records(&self) -> Self::RecordsIter<'_> {
-        let mut records = vec![];
-        for key in self.replication_records.iter() {
-            if let Some(record) = self.get(key) {
-                records.push(record);
-            }
+        RecordsIterator {
+            keys: self.records.iter(),
```

Comment:
> replication_records

I thiiink this could be useful in a follow up PR? Although, that's a PR you're looking at @maqi so let me know what you think